### ISSUE TITLE
Customizable icon for Calendar

### ISSUE THREAD

**Issue by Darkrender** [primefaces/primeng#3509]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

When showing the icon, there is currently no way to style a p-calendar component's button element that contains that icon. This is a big problem because if I need to change the way the input element looks, with bootstrap or custom CSS, I can't modify the way the button looks to match the input element. If it were possible to target the button element with my own component's CSS file, this wouldn't be that big of an issue, but that does not seem to be possible in Angular 4.

This appears to also be an issue with all similar components such as the file upload component that contain a visible button.

TLDR: I humbly request a property input on the p-calendar component that allows me to style the icon button.

Example:
<p-calendar [iconButtonStyleClass]="input-group-addon"></p-calendar>

* **Angular version:** 4.3.1

* **PrimeNG version:** 4.1.1

**Comment by jtdaad** [comment]
this [discussion](https://forum.primefaces.org/viewtopic.php?f=35&t=45607&p=142680#p142680) works
`<p-calendar icon="fa-pencil-square-o"></p-calendar>`

**Comment by vinothbabu** [comment]
Is this issue close? Applying font-awesome icons to icon property, still does not work.

**Comment by muhammadZaahid** [comment]
I need move the icon before the inputtext, does anyone know how?
